service/utils/channel_utils.js

```diff
@@ -98,7 +98,8 @@ function isDirectChannelVisible(userId, myPreferences, channel) {
 
 function isFavoriteChannel(myPreferences, channel) {
     const fav = myPreferences[`${Constants.CATEGORY_FAVORITE_CHANNEL}--${channel.id}`];
-    return fav && fav.value === 'true';
+    channel.isFavorite = fav && fav.value === 'true';
```

Comment:
Oh, I didn't notice that being done anywhere else. Adding extra fields to the model objects that may or may not exist based on how you get them feels like it's going to cause us problems at some point.  That's also going to break the assumption that if an object hasn't changed, then its reference will also remain unchanged.